src/CodeGeneration/OfferFriendlyOverloadsGenerator.cs

```diff
@@ -110,15 +110,23 @@ select new
                     .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
                     .WithExpressionBody(null);
 
-                var flags = GeneratorFlags.NativePointerToIntPtr;
-                var intPtrOverload = transformedMethodBase
-                    .WithParameterList(TransformParameterList(method.ParameterList, flags, parametersToFriendlyTransform))
-                    .WithBody(CallNativePointerOverload(method, flags, parametersToFriendlyTransform));
-                generatedType = generatedType.AddMembers(intPtrOverload);
+                var preexistingMethod = from newMethod in generatedType.Members.OfType<MethodDeclarationSyntax>()
+                                        where WhereIsIntPtrParameter(method.ParameterList.Parameters, newMethod.ParameterList.Parameters).Any() && newMethod.Identifier.ValueText == transformedMethodBase.Identifier.ValueText
```

Comment:
from @AArnott's [comment](https://github.com/AArnott/pinvoke/pull/291#discussion_r80363643):

> This matching criteria seems imprecise enough so as to suppress overload generation in cases where it's not only desirable but already happening, so this would seem to cause API breaking changes since method overloads would disappear.